rust/main/chains/hyperlane-ethereum/src/ism/interchain_security_module.rs

```diff
@@ -116,10 +125,17 @@ where
         message: &HyperlaneMessage,
         metadata: &[u8],
     ) -> ChainResult<Option<U256>> {
-        let tx = self.contract.verify(
+        let mut tx = self.contract.verify(
             metadata.to_owned().into(),
             RawHyperlaneMessage::from(message).to_vec().into(),
         );
+        if self.domain.domain_technical_stack() == HyperlaneDomainTechnicalStack::ZkSync {
```

Comment:
maybe worth mentioning in a comment that since we have duplicate logic for estimating the gas limit, this special casing is also duplicated here: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/cd1d0334411a40950b59efee8b42e341a219cad4/rust/main/chains/hyperlane-ethereum/src/tx.rs#L220

Reply:
Ah good call, added